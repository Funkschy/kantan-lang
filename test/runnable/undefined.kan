import "io";

type MyStruct struct {
    id: i32,
    name: string
}

def main() {
    let ms: MyStruct = undefined;
    init(&ms);

    io.printf("%d %s\n", ms.id, ms.name);
    ms = init_ret(undefined);
    io.printf("%d %s\n", ms.id, ms.name);
}

def init(ms: *MyStruct) {
    ms.id = 1;
    ms.name = "mystruct";
}

def init_ret(ms: MyStruct): MyStruct {
    ms = MyStruct { id: 2, name: "other struct" };
    return ms;
}
