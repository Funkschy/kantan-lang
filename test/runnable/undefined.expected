1 mystruct
2 other struct
